apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/source/GrpcStreamCaller.java

```diff
@@ -0,0 +1,37 @@
+package gov.cms.bfd.pipeline.rda.grpc.source;
+
+import gov.cms.bfd.pipeline.rda.grpc.PreAdjudicatedClaim;
+import io.grpc.ManagedChannel;
+import java.time.Duration;
+import java.util.Iterator;
+
+/**
+ * Interface for objects that handle streaming service calls and map their results to
+ * PreAdjudicatedClaim objects.
+ *
+ * @param <T> Type of objects returned by the streaming service.
+ */
+public interface GrpcStreamCaller<T> {
```

Comment:
This is another one of those things I'll be watching to see how it works out. At the moment, it feels like an odd interface.

For one thing, it's not obvious to me why those two methods are so strongly related that their implementations would always be part of the same interface.

For another -- and this is just a gut feel thing -- it feels a little overly abstracted. Like, this is the kind of thing I'd regret having abstracted myself, a few months/years later. When things _can_ be concrete, they're almost always clearer and more maintainable if you head that route.